packages/xchain-ethereum/src/client.ts

```diff
@@ -317,40 +317,41 @@ export default class Client implements XChainClient, EthereumClient {
    *
    * @throws {"Invalid asset"} throws when the give asset is an invalid one
    */
-  getBalance = async (address?: Address, asset?: Asset): Promise<Balances> => {
+  getBalance = async (address?: Address, assets?: Asset[]): Promise<Balances> => {
     try {
-      address = address || this.getAddress()
-
-      if (asset && assetToString(asset) !== assetToString(AssetETH)) {
-        // Handle token balances
-        const assetAddress = getTokenAddress(asset)
-        if (!assetAddress) {
-          throw new Error('Invalid asset')
-        }
-
-        const balance = await etherscanAPI.getTokenBalance({
-          baseUrl: this.etherscan.baseUrl,
-          address,
-          assetAddress,
-          apiKey: this.etherscan.apiKey,
-        })
-        const decimals = await this.call<BigNumberish>(assetAddress, erc20ABI, 'decimals', [])
-        return [
-          {
-            asset,
-            amount: baseAmount(balance.toString(), BigNumber.from(decimals).toNumber() || ETH_DECIMAL),
-          },
-        ]
-      } else {
-        // Handle ETH balances
-        const balance = await this.etherscan.getBalance(address)
-        return [
-          {
-            asset: AssetETH,
-            amount: baseAmount(balance.toString(), ETH_DECIMAL),
-          },
-        ]
-      }
+      const ethAddress = address || this.getAddress()
+
+      const newAssets = assets || [AssetETH]
+      return Promise.all(
```

Comment:
`Promise.all` will fail if only one request of any balance is failing - try to avoid that.